### PR TITLE
🔨 Add default metrics to Engine

### LINKED ISSUE #1747: 🔨 Add default metric values to `Engine`
### Describe the bug

001_getting_started.ipynb includes the following code to validatate the model:
```
# load best model from checkpoint before evaluating
test_results = engine.test(
    model=model,
    datamodule=datamodule,
    ckpt_path=engine.trainer.checkpoint_callback.best_model_path,
)
```
Printing the `test_results`variable shows that it is empty (`[{}]`).

I have manually checked that the model in question (PaDiM) returns dictionary as part of its Lightning test step. The test step equals to `validation_step` function in the model with 

```
batch["anomaly_maps"] = self.model(batch["image"])
return batch
```
I am not sure where this information gets later lost.

### Dataset

MVTec

### Model

PADiM

### Steps to reproduce the behavior

1. Clone this repo
2. Install anomalib from source using `pip install -e .`, followed by `pip install -e .[full]`.
3. Run notebook "notebooks/001_getting_started.py"

### OS information

OS information:
- OS: macOS and Ubuntu
- Python version: 3.10
- Anomalib version: newest

### Expected behavior

I would expect to see test results either in form of anomaly maps or in form of some metrics already calculated based on the ground truth data. 

### Screenshots

_No response_

### Pip/GitHub

GitHub

### What version/branch did you use?

Newest

### Configuration YAML

```yaml
N/A
```

### Logs

```shell
N/A
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by samet-akcay**
## 📝 Description

- Metric values in Engine is currently `None`. When Engine is used with default values, the validation step is not run since there is no metric. Due to this there is no test results. This PR adds `AUROC` and `F1Score` as the default metrics to both `image` and `pixel` level metrics. 
- 🛠️ Fixes #1747 

## ✨ Changes

Select what type of change your PR is:

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔨 Refactor (non-breaking change which refactors the code base)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [ ] 🔒 Security update

## ✅ Checklist

Before you submit your pull request, please make sure you have completed the following steps:

- [ ] 📋 I have summarized my changes in the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) and followed the guidelines for my type of change (skip for minor changes, documentation updates, and test enhancements).
- [ ] 📚 I have made the necessary updates to the documentation (if applicable).
- [ ] 🧪 I have written tests that support my changes and prove that my fix is effective or my feature works (if applicable).

For more information about code review checklists, see the [Code Review Checklist](../docs/source/markdown/guides/developer/code_review_checklist.md).

### CODE REVIEW

File: src/anomalib/engine/engine.py [hunk 1]
```diff
@@ -139,8 +139,8 @@ def __init__(
         self.normalization = normalization
         self.threshold = threshold
         self.task = TaskType(task)
-        self.image_metric_names = image_metrics
-        self.pixel_metric_names = pixel_metrics
+        self.image_metric_names = image_metrics if image_metrics else ["AUROC", "F1Score"]
+        self.pixel_metric_names = pixel_metrics if pixel_metrics else ["AUROC", "F1Score"]
```
Comment (ashwinvaidya17) [inline src/anomalib/engine/engine.py:143]: Should we have default pixel metrics? What if someone trains a classification task?

**Review by ashwinvaidya17 — COMMENTED** [review]
Thanks just one question

File: src/anomalib/engine/engine.py (hunk 1 above)
Comment (samet-akcay) [reply src/anomalib/engine/engine.py:143]: ah that's a good point!

**Comment by ashwinvaidya17** [comment]
I see that some tests are failing as they directly access `pixel_metric_names`

**Review by ashwinvaidya17 — APPROVED** [review]
Thanks!

**Pull request merged**